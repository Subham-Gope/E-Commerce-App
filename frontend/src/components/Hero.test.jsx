// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import Hero from "./Hero";

// describe("Testing the Hero.jsx file", () => {
//   test("render the hero title, description and image", () => {
//     // rendering and setting snapshot matching
//     const { container } = render(<Hero />);
//     expect(container).toMatchSnapshot();

//     // checking for the title and text
//     const heroTitle = screen.getByText(/latest arrivals/i);
//     expect(heroTitle).toBeInTheDocument();
//     const heroText1 = screen.getByText(/our best seller/i);
//     expect(heroText1).toBeInTheDocument();
//     const heroText2 = screen.getByText(/shop now/i);
//     expect(heroText2).toBeInTheDocument();
//   });

//   //   checking for image
//   const heroImage = screen.getByAltText(/hero image/i);
//   expect(heroImage).toBeInTheDocument();
// });
