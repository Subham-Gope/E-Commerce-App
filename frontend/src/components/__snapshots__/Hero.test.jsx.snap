// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the Hero.jsx file > render the hero title, description and image 1`] = `
<div>
  <div
    class="flex flex-col sm:flex-row border border-gray-400 "
  >
    <div
      class="w-full sm:w-1/2 flex items-center justify-center py-10 sm:py-0"
    >
      <div
        class="text-[#414141]"
      >
        <div
          class="flex items-center gap-2"
        >
          <p
            class="w-8 md:w-11 h-[2px] bg-[#414141]"
          />
          <p
            class="font-medium text-sm md:text-base"
          >
            OUR BEST SELLER
          </p>
        </div>
        <h1
          class="prata-regular text-3xl sm:py-3 lg:text-5xl leading-relaxed"
        >
          Latest Arrivals
        </h1>
        <div
          class="flex items-center gap-2"
        >
          <p
            class="font-semibold text-sm md:text-base"
          >
            SHOP NOW
          </p>
          <p
            class="w-8 md:w-11 h-[2px] bg-[#414141]"
          />
        </div>
      </div>
    </div>
    <img
      alt="Hero Image"
      class="w-full sm:w-1/2"
      src="/src/assets/hero_img.png"
    />
  </div>
</div>
`;
